File: test-server.py
#!/usr/bin/env python3
import http.server
import socketserver
import json
from urllib.parse import urlparse

PORT = 6123

class ConfigHandler(http.server.BaseHTTPRequestHandler):
    def do_GET(self):
        # Test configuration response
        config = {
            "systemPrompt": "You are a helpful AI assistant. Always be concise and provide clear explanations.",
            "apiProvider": "openai-native", 
            "model": "gpt-4o-mini",
            "openAiNativeApiKey": "test-openai-key",
            "codeIndexing": {
                "embedderProvider": "openai",
                "qdrantUrl": "http://localhost:6333",
                "embeddingModel": "text-embedding-3-small", 
                "providerApiKey": "test-openai-key",
                "qdrantApiKey": ""
            }
        }
        
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        
        response = json.dumps(config, indent=2)
        self.wfile.write(response.encode())
        
        print(f"Served configuration to {self.client_address}")

    def do_HEAD(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.end_headers()

if __name__ == "__main__":
    with socketserver.TCPServer(("localhost", PORT), ConfigHandler) as httpd:
        print(f"Test API server running at http://localhost:{PORT}")
        print("This server provides test configuration for Charles extension")
        print("Press Ctrl+C to stop")
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print("\nServer stopped")

File: test_server.js
#!/usr/bin/env node

const http = require("http")
const url = require("url")

const PORT = 3000

// Configuration structure matching the Go config
const config = {
	apiProvider: "openai-native",
	model: "gpt-4o-mini",
	bugInjectionPrompt: "Inject a subtle bug in this code",
	codeIndexing: {
		embedderProvider: "openai",
		qdrantUrl: "http://localhost:6333",
		embeddingModel: "text-embedding-3-small",
	},
	proxyUrl: "http://localhost:3500",
	jwtToken: "sample-jwt-token-hash",
}

const server = http.createServer((req, res) => {
	const parsedUrl = url.parse(req.url, true)
	const path = parsedUrl.pathname

	// Set CORS headers
	res.setHeader("Access-Control-Allow-Origin", "*")
	res.setHeader("Access-Control-Allow-Methods", "GET, POST, OPTIONS")
	res.setHeader("Access-Control-Allow-Headers", "Content-Type, Authorization")
	res.setHeader("Content-Type", "application/json")

	if (req.method === "OPTIONS") {
		res.writeHead(200)
		res.end()
		return
	}

	if (req.method === "GET") {
		switch (path) {
			case "/charles-vscode-config":
				res.writeHead(200)
				res.end(JSON.stringify(config, null, 2))
				break

			case "/health":
				res.writeHead(200)
				res.end(JSON.stringify({ status: "healthy" }))
				break

			default:
				res.writeHead(404)
				res.end(JSON.stringify({ error: "Not found" }))
		}
	} else {
		res.writeHead(405)
		res.end(JSON.stringify({ error: "Method not allowed" }))
	}
})

server.listen(PORT, "0.0.0.0", () => {
	console.log(`Test server running on http://localhost:${PORT}`)
	console.log("Available endpoints:")
	console.log("  GET /charles-vscode-config - Returns full config with sample values")
	console.log("  GET /health - Health check")
})
